File: Solution_Practice_I/java/prob2_5.java
import java.util.Scanner;
import java.time.LocalDate;
public class prob2_5 {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        LocalDate actualDate= LocalDate.now();
        System.out.print(actualDate);
        int day, month, year, available;
        do {
            System.out.println("Ingrese la fecha");
            System.out.print("Dia: ");
            day=input.nextInt();
            System.out.print("Mes: ");
            month=input.nextInt();
            System.out.print("Año: ");
            year=input.nextInt();
        } while (year<0 || month<=0 || day<=0);
        if (month>12) available=0;
        else if (month == 2) {
            if ((year % 4 == 0 && year % 100 > 0) || (year % 400 == 0)) available=29;
            else available=28;            
        }
        else if(
            month==4 || 
            month==6 ||
            month==9 ||
            month==11 
        ) available = 30;
        else available = 31;
        if (day<=available) {
            LocalDate enteredDate = LocalDate.of(year, month, day);
            if(!enteredDate.isAfter(actualDate)) System.out.println("Fecha Correcta");
            else System.out.println("Fecha Incorrecta");
        }else System.out.print("Fecha Incorrecta");
        input.close();
    }
}

File: Solution_Practice_I/python/prob2_5.py
from datetime import date

while True:
    print(f"Ingrese la fecha → (Día Mes Año): ")
    day, month, year = map(int, input().split())
    if 1 <= month and 1 <= day <= 31 and year >= 0: break
    else: print("Fecha Incorrecta")

if month>12: 
    available=0
elif month == 2:
    available = 29 if (year % 4 == 0 and year % 100 != 0) or (year % 400 == 0) else 28
elif month in [4, 6, 9, 11]:
    available = 30
else:
    available = 31

if day <= available:
    entered_date = date(year, month, day) #202
    if entered_date <= date.today():
        print("Fecha Correcta")
    else:
        print("Fecha Incorrecta")
else:
    print("Fecha Incorrecta")
